sys/net/link_layer/lorawan/Kconfig

```diff
@@ -0,0 +1,476 @@
+# Copyright (c) 2020 Freie Universitaet Berlin
+#
+# This file is subject to the terms and conditions of the GNU Lesser
+# General Public License v2.1. See the file LICENSE in the top level
+# directory for more details.
+#
+menuconfig KCONFIG_USEMODULE_LORAWAN
+    bool "Configure LoRaWAN MAC"
+    depends on USEMODULE_LORAWAN
```

Comment:
I'd do something different here. Instead of introducing a new module only to show the lorawan options, it could be done with a symbol that is selected by the used LoRaWAN stack. An example of this is the CoAP generic options:
https://github.com/RIOT-OS/RIOT/blob/aabd5eaca50e2a59f3dcc0eeb3b75940647a9d45/sys/net/application_layer/Kconfig.coap#L13-L15

`HAS_PROTOCOL_COAP` is selected by nanocoap, which accepts those configuration symbols. Something similar could be with GNRC LoRaWAN and Semtech, both could select this non-visible symbol. Although I'd suggest to use the prefix `HAVE_`, as `HAS_` is now reserved for another used.

Alternatively, it could be replaced by
```
depends on USEMODULE_GNRC_LORAWAN || USEPKG_SEMTECH_LORAMAC
```
If we don't expect to have many others LoRaWAN stacks...

Reply:
Thanks for the suggestion. `HAS_PROTOCOL_COAP` requires `SELECT`. I guess that's okay here. Shall update it :)

Reply:
Yes, it is a non-visible symbol, it is used to indicate certain conditions. But if you will need to include the stack symbols anyways to show or hide other configuration parameters down on the file, maybe there is no need for this extra symbol, and you can go with `depends on USEMODULE_GNRC_LORAWAN || USEPKG_SEMTECH_LORAMAC`

Reply:
Yes. `depends on USEMODULE_GNRC_LORAWAN || USEPKG_SEMTECH_LORAMAC` will make better sense.